llm_studio/src/datasets/text_causal_language_modeling_ds.py

```diff
@@ -294,12 +301,17 @@ def _get_sample(self, idx):
                 ],
                 dim=0,
             )
-        return [prompt_encodings, answer_encodings]
+
+        if self.cfg.training.use_rlhf and self.mode == "train":
+            return [prompt_encodings]
+        else:
+            return [prompt_encodings, answer_encodings]
```

Comment:
Answer encodings would still be needed for the chat history (when using `parent_id_column`)?

Reply:
In this case, afaik, the "old" answers are in the prompt. But it's definetly tricky, as they need to be passed to the reward model, too. Which is not really trained on full conversations but rather single question, single answer if I see that correctly.